inc/classes/subscriber/Addons/Cloudflare/CloudflareSubscriber.php

```diff
@@ -90,4 +123,418 @@ private function should_filter_varnish() {
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge Cloudflare cache URLs automatically if Cache Everything is set as a Page Rule
+	 *
+	 * @since 3.4.2
+	 * @author Soponar Cristina
+	 *
+	 * @param WP_Post $post       The post object.
+	 * @param array   $purge_urls URLs cache files to remove.
+	 * @param string  $lang       The post language.
+	 */
+	public function auto_purge_by_url( $post, $purge_urls, $lang ) {
+		if ( ! $this->options->get( 'do_cloudflare' ) || ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		$cf_cache_everything = $this->cloudflare->has_page_rule( 'cache_everything' );
+
+		if ( is_wp_error( $cf_cache_everything ) || ! $cf_cache_everything ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->cloudflare->purge_by_url( $post, $purge_urls, $lang );
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge CloudFlare cache
+	 *
+	 * @since 2.5
+	 */
+	public function do_purge_cloudflare() {
+		if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( sanitize_key( $_GET['_wpnonce'] ), 'rocket_purge_cloudflare' ) ) {
+			wp_nonce_ays( '' );
+		}
+		if ( ! $this->options->get( 'do_cloudflare' ) || ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->cloudflare->purge_cloudflare();
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+
+		wp_safe_redirect( esc_url_raw( wp_get_referer() ) );
+		die();
```

Comment:
`wp_die();` is not going to work with `wp_safe_redirect()`. Instead of the redirect an empty error page will be displayed and this is not the desired result. 
eg. with `wp_die();` - https://www.screencast.com/t/lFFacV6QU6w
eg. without any die or wp_die it works fine: so I think I can remove totally the die from this part

Reply:
For security reasons, `die();` must not be removed.

Reply:
Indeed it's important to keep it. In that case, the alternative is to keep `die();`, and move the  rest of the method's code to a different method, which will be testable.

Reply:
Done